### PR TITLE
update to alpine3.5 and nginx 1.10.2

### PR DESCRIPTION
**Pull request by stefancocora**
- the nginx version 1.10.2 in alpine3.5 comes with the
http_realip_module module compiled in. This module is used to
whitelist/blacklist on proxyprotocol set ips from upstream
loadbalancers.

### CODE REVIEW

**Comment by vaijab** [comment]
> nginx version 1.10.2

Any breaking changes?

**Comment by stefancocora** [comment]
No breaking changes as far as I've seen, although it is really hard to ascertain without a list of current supported features.
Do you have this list of currently supported features for this nginx package ?

Semver wise I'll bump this to `v1.0.0` just because it a newer version of nginx and the alpine maintainers have enabled a lot more nginx modules at compile time compared to alpine3.4's nginx
http://git.alpinelinux.org/cgit/aports/tree/main/nginx/APKBUILD?h=3.5-stable

**Comment by vaijab** [comment]
I guess an upgrade makes sense, but I'd like more testing to be done before any release. Can you build an image of this branch and use it for testing? You mentioned you have a real use-case that needs certain features from the upgrade.

**Comment by stefancocora** [comment]
> Can you build an image of this branch and use it for testing? 

This is easy , I can build a test image.

> You mentioned you have a real use-case that needs certain features from the upgrade.

This isn't as trivial as you'd think, the feature I need is in the reverse proxy part of nginx in the real_ip header module.
To test this part I'd need to bring up a testing stack that looks like 
- ` HTTP caller code ==> this_docker_container_in_reverse_proxy ==> backend webserver `
- `capture the result at the "backend webserver" and parse the headers`

While this can be done it is costly and time consuming, think of the project wait times too please.

Happy to go ahead with this testing routine if you consider it absolutely necessary. It is your call !

**Comment by vaijab** [comment]
> Happy to go ahead with this testing routine if you consider it absolutely necessary

I was thinking in terms of testing in general. If you can build a test image and use it in real world, that'd be quite good to verify that the upgrade does not break many things.

**Comment by stefancocora** [comment]
Now it is clear what you meant earlier with testing.
I'll build an image from the branch changes and test it as a replacement for docker-proxy-nginx.
I'll put my findings in this ticket.

**Comment by vaijab** [comment]
> I'll put my findings in this ticket.

Perfecto! There are no tests here, so I am just a little worried, that's all :-) It's the nginx version upgrade, but I am more worried that alpine now compiles in every single module that there is.

**Comment by stefancocora** [comment]
I've tested the nginx image `845ec912426f121580b17c07a409e8b860209e6b` with nginx `v1.10.2` with real apps on the dev cluster.
These nginx modules ( and probably others that I can't remember ) have been exercised in this test and work fine:
- nginx core module
- http://nginx.org/en/docs/http/ngx_http_log_module.html
- http://nginx.org/en/docs/ngx_core_module.html#error_log
- https://nginx.org/en/docs/stream/ngx_stream_proxy_module.html#proxy_protocol
- http://nginx.org/en/docs/http/ngx_http_access_module.html

**Comment by vaijab** [comment]
@stefancocora awesome. Please clean up drone.yaml and this is good to go then! Thanks!

**Comment by stefancocora** [comment]
`.drone.yml` has been cleaned up

**Comment by vaijab** [comment]
> .drone.yml has been cleaned up

Not sure you have. You deleted push master event.

**Comment by stefancocora** [comment]
Something went wrong with my rebase, I'll re-do it.

**Comment by stefancocora** [comment]
done, my local branch was detached from the upstream branch that I pushed earlier.

**Review by vaijab — APPROVED** [review]
Great, thanks. Don't release it yet. CI will build a new image off master anyway if you must use this.

**Comment by stefancocora** [comment]
Ok I'll merge it so that CI bulds an image that I can use straight away.
Any reason why you don't want to release it straight away ?

**Pull request merged**
